cmake/HPX_SetupCUDA.cmake

```diff
@@ -6,57 +6,118 @@
 
 if(HPX_WITH_CUDA AND NOT TARGET Cuda::cuda)
 
-  find_package(CUDA REQUIRED)
+  if(HPX_WITH_CUDA_CLANG AND NOT (CMAKE_CXX_COMPILER_ID STREQUAL "Clang"))
+    hpx_error(
+      "To use Cuda Clang, please select Clang as your default C++ compiler"
+    )
+  endif()
+
+  # cuda_std_17 not recognized for previous versions
+  cmake_minimum_required(VERSION 3.18 FATAL_ERROR)
```

Comment:
I think we should just require this for all configurations. However, it does mean another CMake version bump. @hkaiser, @biddisco, objections?

Reply:
No objections.